homeassistant/components/shelly/sensor.py

```diff
@@ -991,20 +991,18 @@ class RestSensorDescription(RestEntityDescription, SensorEntityDescription):
     ),
     "counter_frequency": RpcSensorDescription(
         key="input",
-        sub_key="counts",
+        sub_key="freq",
         name="Pulse counter frequency",
         native_unit_of_measurement=UnitOfFrequency.HERTZ,
         state_class=SensorStateClass.MEASUREMENT,
-        value=lambda status, _: status["freq"],
-        removal_condition=lambda config, status, key: (config[key]["enable"] is False),
+        removal_condition=lambda config, _, key: (config[key]["enable"] is False),
```

Comment:
Shouldn't we test for get('freq') is None like we do for xfreq ?

Reply:
According to the documentation `freq` is always a number
![obraz](https://github.com/home-assistant/core/assets/478555/99adf272-2067-4f60-aa1a-4afc3e7967bb)
But I think we need here to check the type of the input. If I understand correctly `freq` is available only for the type `count`
https://shelly-api-docs.shelly.cloud/gen2/ComponentsAndServices/Input#status

Reply:
yup, `percent `and `xpercent` for **analog**, `freq` and `xfreq` for **count**